### DISCUSSION TITLE
How does GitHub assign numbers to issues, pull requests and discussions?

### DISCUSSION THREAD

**Discussion by shenweiyan** [community/community#69759] [Discussions] [answered]
### Select Topic Area

Question

### Body

As for this [issue](https://stackoverflow.com/questions/72042182/how-does-github-assign-numbers-to-issues-pull-requests-and-discussions) on stackoverflow:
> I am quite sure that GitHub uses the same counter for issues, pull requests and discussions.
> 
> If I create an issue in a fresh new repository, its number would be #1. Then, if I want to create a pull request, the number assigned to it would be #2. Similarly if I then create a discussion, its number would be #3.
> 
> I just wanted to get some sort of confirmation for my guess but couldn't find any in the GitHub docs.

Can someone confirm this? Who can explain why GitHub is designed the way it is?

**Comment by mdavic0** [comment, marked as answer]
GitHub uses a single, shared counter for issues, pull requests, and discussions within a repository. This means that when you create a new item (be it an issue, pull request, or discussion) in a fresh repository, the numbering starts at 1 and increments for each subsequent item, regardless of its type. Therefore, the numbering sequence is unified across these three categories of items within a repository.

This design simplifies the process of numbering and referencing items within a repository. It provides a consistent and straightforward way to identify and link to specific items using a common numbering system.

While this behavior is consistent and well-documented, it's not explicitly explained in the GitHub documentation because it's a fundamental part of how GitHub organizes and tracks items within repositories. The shared numbering system is a part of GitHub's design for providing a unified issue-tracking system that includes issues, pull requests, and discussions

**Reply by zorana-gavrilovic** [reply]
Hello, does the numbering logic also apply to issue comments? Or rather, will every next issue comment have a higher ID than the previous one? 
Best regards
